src/allotropy/parsers/unchained_labs_lunatic/unchained_labs_lunatic_structure.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Optional
+import uuid
+
+import pandas as pd
+
+from allotropy.exceptions import AllotropeConversionError
+from allotropy.parsers.unchained_labs_lunatic.constants import (
+    INCORRECT_WAVELENGHT_COLUMN_FORMAT_ERROR_MSG,
+    NO_DATE_OR_TIME_ERROR_MSG,
+    NO_MEASUREMENT_IN_PLATE_ERROR_MSG,
+    NO_WAVELENGHT_COLUMN_ERROR_MSG,
+    WAVELENGHT_COLUMNS_RE,
+)
+from allotropy.parsers.utils.values import try_float_from_series, try_str_from_series
+
+
+@dataclass(frozen=True)
+class Measurement:
+    measurement_identifier: str
+    wavelenght: float
+    absorbance: float
+    sample_identifier: str
+    location_identifier: str
+    well_plate_identifier: Optional[str]
+
+    @staticmethod
+    def create(well_plate_data: pd.Series[Any], wavelenght_column: str) -> Measurement:
+        if wavelenght_column not in well_plate_data:
+            msg = NO_MEASUREMENT_IN_PLATE_ERROR_MSG.format(wavelenght_column)
+            raise AllotropeConversionError(msg)
+
+        if not WAVELENGHT_COLUMNS_RE.match(wavelenght_column):
+            raise AllotropeConversionError(INCORRECT_WAVELENGHT_COLUMN_FORMAT_ERROR_MSG)
+
+        wavelenght = float(wavelenght_column[1:])
+
+        return Measurement(
+            measurement_identifier=str(uuid.uuid4()),
+            wavelenght=wavelenght,
+            absorbance=try_float_from_series(well_plate_data, wavelenght_column),
+            sample_identifier=try_str_from_series(well_plate_data, "Sample name"),
+            location_identifier=try_str_from_series(well_plate_data, "Plate ID"),
+            well_plate_identifier=well_plate_data.get("Plate Position"),  # type: ignore[arg-type]
+        )
+
+
+@dataclass(frozen=True)
+class WellPlate:
+    measurement_time: str
+    analytical_method_identifier: Optional[str]
+    measurements: list[Measurement]
+
+    @staticmethod
+    def create(plate_data: pd.Series[Any], wavelenght_columns: list[str]) -> WellPlate:
+        return WellPlate(
+            measurement_time=WellPlate.get_datetime_from_plate(plate_data),
+            analytical_method_identifier=plate_data.get("Application"),  # type: ignore[arg-type]
+            measurements=[
+                Measurement.create(plate_data, wavelenght_column)
+                for wavelenght_column in wavelenght_columns
+            ],
+        )
+
+    @staticmethod
+    def get_datetime_from_plate(plate_data: pd.Series[Any]) -> str:
```

Comment:
This should only be called by `create`, so let's protect this -- e.g., `_get_datetime_from_plate`.